proxy/apiserver/router/router.go

```diff
@@ -111,6 +120,74 @@ func (r *router) Route(httpReq *http.Request, reqInfo *request.RequestInfo) (*Ro
 	return &RouteAccept{}, nil
 }
 
+func (r *router) injectSelector(httpReq *http.Request, gvr *schema.GroupVersionResource, snapshot *proxyclient.ProtoSpecSnapshot) error {
+	protoSpec, _, err := snapshot.AcquireSpec()
+	if err != nil {
+		return err
+	}
+	var subset *proto.InternalSubsetLimit
+	for _, sub := range protoSpec.RouteInternal.SubsetLimits {
+		if sub.Subset == protoSpec.RouteInternal.Subset {
+			subset = sub
+			break
+		}
+	}
+	if subset == nil {
+		return fmt.Errorf("can not find subset rule %s", protoSpec.RouteInternal.Subset)
+	}
+	lbSelector := mergeSelector(determineObjectSelector(subset.Limits, gvr), determineObjectSelector(protoSpec.RouteInternal.GlobalLimits, gvr))
+	selector, err := metav1.LabelSelectorAsSelector(lbSelector)
+	if err != nil {
+		return err
+	}
+	raw, err := httputil.ParseRawQuery(httpReq.URL.RawQuery)
+	if err != nil {
+		return err
+	}
+	raw[labelSelectorKey] = url.QueryEscape(selector.String())
```

Comment:
Should we check if there already exist label selector in the query?

Reply:
If operator uses controller-runtime manager, it list&watch without selector. But what should i do if there're already exist label selector in the query when operator uses restClient list&watch directly? Merge? Replace? Or ignore?

Reply:
I'm not sure... Maybe merge is better?

> If operator uses controller-runtime manager, it list&watch without selector.

Since controller-runtime v0.9.0, it provides a `SelectorsByObject` to support label/field selector for listwatch.
https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/cache/cache.go#L118